docs/design-docs/benchmarks.md

```diff
@@ -0,0 +1,274 @@
+---
+id: benchmarks
+---
+
+# Benchmarks Design
+
+## Goals
+
+The design of benchmarks has the following goals to achieve:
+
+**High Code Quality**
+* Extract the common code into the base class, reduce the workload to maintain all benchmarks.
```

Comment:
reduce the efforts to maintain different benchmarks.

Reply:
Done.